saleor/graphql/checkout/mutations.py

```diff
@@ -480,17 +480,17 @@ def perform_mutation(cls, _root, info, checkout_id, shipping_address):
 
         cleaned_address_data = cls.validate_address(shipping_address)
 
-        checkout.shipping_address = cls.construct_instance(
-            checkout.shipping_address or Address(),
+        shipping_address = cls.construct_instance(
+            shipping_address,
```

Comment:
`construct_instance` gets `instance` of type `Address` here as a first parameter, but you provides `AddressInput`.